hdk-rust/wasm-test/src/lib.rs

```diff
@@ -333,6 +333,13 @@ fn hdk_test_entry() -> Entry {
     Entry::App(hdk_test_app_entry_type(), hdk_test_entry_value())
 }
 
+fn handle_send_message(to_agent: Address, message: String) -> String {
```

Comment:
So `send` and receive can pass/return String, can it use other types?

Like could `receive` return a ZomeApiResult?

Reply:
I tried with a `Result<String, String>` but discovered a bug in `JsonString::from<Result<>>`. Already synced with @thedavidmeister. So I reverted back to just String for now.

Basically possible but tried to keep the scope small for this one.

Reply:
I see, ok well knowing that we will implement a complex struct, not just a string eases my mind on this one 👍

Reply:
oh yeah i forgot about that bug

i'd better actually fix that! :sweat_smile:

Reply:
Is there a follow up created for this one @lucksus @thedavidmeister ?